src/map/script.c

```diff
@@ -6668,18 +6668,20 @@ static BUILDIN(warpchar)
 	x=script_getnum(st,3);
 	y=script_getnum(st,4);
 	a=script_getnum(st,5);
+	unsigned short map_index;
 
 	sd = script->charid2sd(st, a);
 	if (sd == NULL)
 		return true;
 
-	if(strcmp(str, "Random") == 0)
+	if (strcmp(str, "Random") == 0)
 		pc->randomwarp(sd, CLR_TELEPORT);
+	else if (strcmp(str, "SavePoint") == 0)
+		pc->setpos(sd, sd->status.save_point.map,sd->status.save_point.x, sd->status.save_point.y, CLR_TELEPORT);
+	else if ((map_index = script->mapindexname2id(st, str)) != 0)
+		pc->setpos(sd, map_index, x, y, CLR_TELEPORT);
 	else
-		if(strcmp(str, "SavePoint") == 0)
```

Comment:
Can you please revert back this change, only change the call in this commit and apply the code improvement in another commit.